Tweak: Few interactions for cyborg grippers

fake123acc opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
Гриппер инженерного и медицинского борга теперь позволяют применять инструменты на предмет, который находится в гриппере.
Примеры: настройка платы вендомата (требует применения отвертки, пока плата в руке, ранее инженерный борг мог держать плату, но не мог её настроить), потрошение органов (требовали от медборга выложить орган на пол).
В проверку на нахождение платы в руке для вендомата добавлено условие на нахождение в гриппере борга, который применяет отвёртку.

_Не проходило предложку, автор сомневается в наличии неожиданных, не покрытых тестированием вариантов, однако они возможны. По этой причине ПР в драфте до:_
_1) получения разрешения на мёрж/прохождение воута_ **СДЕЛАНО**
_2) дополнительного ревью и тестирования на предмет нежелательных взаимодействий_ **СДЕЛАНО, НАСКОЛЬКО ВОЗМОЖНО**

upd: предложка прошла, https://discord.com/channels/617003227182792704/755125334097133628/1007964492577120317

## Why It's Good For The Game
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Предсказуемость поведения игры. Допустимые взаимодействия приносят ожидаемый результат.

## Changelog
:cl:
tweak: функционал гриппера борга дополнен взаимодействиями с инструментами.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

perryjoy committed:
engineering cyborg gripper & vendor circuit board fixes

+added tool_act for engineering cyborg gripper (calls tool_act of held object)

+ added gripper check for vendor circuit board besides loc check (for detecting if the circuit board is in hands)

perryjoy committed:
added attackby proc for item held in gripper

now medical cyborg is able to butcher an organ without dropping it on the floor

PhantornRU commented:
По коду не заметил недочетов. Работает? Работает.
От себя лично я бы скриншоты еще рекомендовал делать

fake123acc committed:
added check for item destruction

fixed item pointer being stored and gripper was considered as occupied even after item itself was dissasmbled (i.e. wrenching the APC frame while holding it with gripper)

fake123acc commented:
> От себя лично я бы скриншоты еще рекомендовал делать

Там разве что скриншоты чатлога, что плата вендора меняется. Мхех, такое себе.

Добавил проверку, без которой разбор АПЦ фрейма, находившегося в лапке, приводил к фантомному фрейму, который не давал взять предмет и требовал прожать Q (т.к. он помечен как удалённый, фрейм не выпадал, несмотря на сообщение в чат "You drop the APC frame")

Vallat commented on code/modules/mob/living/silicon/robot/drone/drone_items.dm:
```diff
@@ -82,6 +82,18 @@
 		to_chat(user, "<span class='warning'>[src] is empty.</span>")
 
+/obj/item/gripper/tool_act(mob/living/user, obj/item/tool, tool_type)
+	if(gripped_item)
```
Друг, ты `early returns` обронил 🤓

fake123acc replied:
Это другое! :Kappa:

Vallat replied:
> Это другое! :Kappa:

Не, не другое 🤓 
То, что меня нет, не значит, что кодстандарты были отменены

fake123acc replied:
Подожди, я думал это ты к тому, что я топил за early return'ы когда в моём макросе для токсинов ты убрал проверку, завернув код в if
Тут я хотел быть последовательным: в соседних функциях аналогичный синтаксис

Vallat replied:
Вот макросы это как раз другое

Vallat commented on code/modules/mob/living/silicon/robot/drone/drone_items.dm:
```diff
@@ -82,6 +82,18 @@
+/obj/item/gripper/tool_act(mob/living/user, obj/item/tool, tool_type)
+	if(gripped_item)
+		gripped_item.tool_act(user, tool, tool_type)
+		if (gripped_item.gc_destroyed) // if item was dissasembled we need to clear the pointer
```
```suggestion
		if (QDELETED(gripped_item)) // if item was dissasembled we need to clear the pointer
```

fake123acc replied:
Научи, чем одно лучше другого?

Vallat replied:
> Научи, чем одно лучше другого?

Всегда лучше использовать специализированные макросы, чем просто повторять их функционал. Ведь когда-нибудь механизм определения удалён ли объект может измениться и вместо изменения подобной проверки по всему проекту мы просто поменяем логику одного макроса и всё

Vallat commented on code/modules/mob/living/silicon/robot/drone/drone_items.dm:
```diff
@@ -82,6 +82,18 @@
+/obj/item/gripper/attackby(obj/item/weapon, mob/user, params)
+	if(gripped_item)
+		gripped_item.attackby(weapon, user, params)
+		if (gripped_item.gc_destroyed) // if item was dissasembled we need to clear the pointer
```
```suggestion
		if (QDELETED(gripped_item)) // if item was dissasembled we need to clear the pointer
```

fake123acc committed:
gc_destroyed into QDELETED change 1

Vallat says its better so its better I guess

Co-authored-by: Vallat <vox.crit2013@gmail.com>

fake123acc committed:
gc_destroyed into QDELETED change 2

Vallat says its better so its better I guess

Co-authored-by: Vallat <vox.crit2013@gmail.com>

fake123acc commented:
Предложка прошла, прошу ещё раз глянуть на код на предмет неожиданных абузов, если нареканий нет, пихну ранние ретёрны по настоянию Валлата и вывожу из драфта

fake123acc committed:
Early returns 4 Vallat

Fixed my code and drop_gripped_item proc nearby

fake123acc commented:
Я закончил, прошу проставить галочки для благополучного мёржа
